### PR TITLE
Update magento2 nginx to use https

### PR DESCRIPTION
**Pull request by kierenevans**
Allow improvements to the nginx config in pho-nginx to shine through in this image by not overwriting the whole virtual host.

This brings us:
* Automatic HTTPS
* Automatic redirect to HTTPS from HTTP
* Automatic HTTP2
* Any future benefits!

I need to test this.

### CODE REVIEW

File: magento2-nginx/fpm-7.0/etc/nginx/sites-available/default-50_site_final_match.conf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Banned locations (only reached if the earlier PHP entry point regexes don't match)
+location ~* (\.php$|\.htaccess$|\.git) {
```
Comment (andytson-inviqa) [inline magento2-nginx/fpm-7.0/etc/nginx/sites-available/default-50_site_final_match.conf]: `(/|^)([^\]*.php|\.(ht|git)[^\]*)(/|$)` is more secure (e.g. blocks /index.php/something and doesn't block /somethingabout.git/.

Comment (kierenevans) [inline magento2-nginx/fpm-7.0/etc/nginx/sites-available/default-50_site_final_match.conf]: I shall update the template.

Comment (andytson-inviqa) [inline magento2-nginx/fpm-7.0/etc/nginx/sites-available/default-50_site_final_match.conf]: updated comment to \.php :)

Comment (andytson-inviqa) [inline magento2-nginx/fpm-7.0/etc/nginx/sites-available/default-50_site_final_match.conf]: sorry some / were \\

File: magento2-nginx/fpm-7.0/usr/local/share/env/30-framework [hunk 2]
```diff
@@ -4,7 +4,7 @@ export PHP_MEMORY_LIMIT=${PHP_MEMORY_LIMIT:-768M}
 
 export PRODUCTION_ENVIRONMENT=${PRODUCTION_ENVIRONMENT:-false}
 export APP_HOSTNAME=${APP_HOSTNAME:-magento_web.docker}
-export PUBLIC_ADDRESS=${PUBLIC_ADDRESS:-http://magento_web.docker/}
+export PUBLIC_ADDRESS=${PUBLIC_ADDRESS:-https://magento_web.docker/}
```
Comment (andytson-inviqa) [inline magento2-nginx/fpm-7.0/usr/local/share/env/30-framework]: according to the readme you shouldn't default to the _ here

**Pull request merged**